src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -180,21 +180,52 @@ function validatePropTypes(element) {
   }
 }
 
+function warnInvalidType(inputValue) {
+  // warning(...) messages need to be string literals.
+  // If not, we can do a ternary expression to evaluate the need
+  // for the mistyped import message, and not repeat the message
+  // twice here.
+  if (inputValue === undefined) {
+    warning(
+      false,
+      'React.createElement: %s is an invalid value for a type. ' +
```

Comment:
We already know it is undefined, no need to interpolate it. Let’s just hardcode it to make it clear.